docs/sections/negotiation_guide.ipynb

```diff
@@ -0,0 +1,465 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "\n",
+    "# Step-by-step agent negotiation\n",
+    "\n",
+    "\n",
+    "In this section, you will see how the negotiation works in our framework.\n",
+    "\n",
+    "## The big picture\n",
+    "\n",
+    "\n",
+    "The following diagram gives you a high-level understanding on how a generic negotiation works.\n",
+    "\n",
+    ".. mermaid:: ../diagrams/negotiation-guide/summary.mmd\n",
+    "\n",
+    "Let's see step by step what happens:\n",
+    "\n",
+    "1. `Agent_1` calls `get_service_description(is_supply)` to generate the service description. `is_supply` is a flag to\n",
+    "    switch between the seller and buyer roles.\n",
+    "2. `Agent_1` sends a `register_service` request to the OEF node, so she can be advertised on the OEF.\n",
+    "3. Analogous to (1), but for `Agent_2`\n",
+    "4. Analogous to (2), but for `Agent_2`\n",
+    "5. `Agent_1` calls `build_services_query(is_searching_for_sellers)` to generate a `query` for the OEF\n",
+    "6. `Agent_1` send a `search_service` request with the `query` previously generated.\n",
+    "7. The OEF node returns a search result with the list of agent ids matching the `query`\n",
+    "8. `Agent_1` finds `Agent_2`, so `Agent_1` sends a `CFP` to `Agent_2`, meaning that she wants to start a negotiation.\n",
+    "9. `Agent_2` calls `get_proposals()` to generate a proposal for answering `Agent_1`\n",
+    "10. `Agent_2` replies with a `Propose` message as an answer for the `CFP`.\n",
+    "11. `Agent_1` sends an `Accept` message to `Agent_2`, meaning that she accepts the proposal.\n",
+    "12. `Agent_1` sends a Transaction request to the `Controller`.\n",
+    "13. `Agent_2` replies with a _matched accept_ to `Agent_1`, meaning that she confirms definitively the transaction. \n",
+    "14. `Agent_2` sends a Transaction request to the `Controller` (analogous to step 12).\n",
+    "\n",
+    "\n",
+    "Notice that this is the behaviour of the `BaselineAgent`. By modifying the default strategy, you can change\n",
+    "the behaviour in steps 1 (or 3), 5 and 9. The other methods are handled by the `tac` .\n",
```

Comment:
Suggest to change:
..are handled by the `tac`.
>>
..are handled by our implementation of the FIPA negotiation protocol.